flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -434,16 +460,19 @@ public CompletableFuture<JobID> submitJob(@Nonnull JobGraph jobGraph) {
                                     });
                         });
 
-        submissionFuture
-                .thenCompose(ignored -> jobGraphFileFuture)
-                .thenAccept(
-                        jobGraphFile -> {
-                            try {
-                                Files.delete(jobGraphFile);
-                            } catch (IOException e) {
-                                LOG.warn("Could not delete temporary file {}.", jobGraphFile, e);
-                            }
-                        });
+        submissionFuture.whenComplete(
+                (i1, i2) ->
+                        jobGraphFileFuture.thenAccept(
+                                jobGraphFile -> {
+                                    try {
+                                        Files.delete(jobGraphFile);
+                                    } catch (IOException e) {
+                                        LOG.warn(
+                                                "Could not delete temporary file {}.",
+                                                jobGraphFile,
+                                                e);
+                                    }
+                                }));
```

Comment:
nit:
this seems a bit more readable to me:
```
        submissionFuture
                .exceptionally(ign -> null)   // ignore errors
                .thenAccept(
                        jobGraphFile -> {
...
```

Reply:
`exceptionally` simplifies the change quite a bit, added.